common-tasks/usb.md

```diff
@@ -21,199 +21,108 @@ redirect_from:
+the partition:
+    ```
+    cd ~
+    mkdir mnt
+    sudo mount /dev/xvdi mnt
+    ```
+    And when done:
+    `sudo umount mnt`
+    
+The command-line tool you may use to mount whole USB drives or their partitions
+is `qvm-block`. This tool can be used to assign a USB drive to a qube as
+follows:
 
-In order to use a USB keyboard, you must first attach it to a USB qube, then give that qube permission to pass keyboard input to dom0.
-Edit the `qubes.InputKeyboard` policy file in dom0, which is located here:
+ 1. Insert your USB drive.
 
-    /etc/qubes-rpc/policy/qubes.InputKeyboard
+ 2. In a dom0 console (running as a normal user), list all available block
+    devices:
 
-Add a line like this one to the top of the file:
+        qvm-block
 
-    sys-usb dom0 allow,user=root
+    This will list all available block devices connected to any USB controller
+    in your system, no matter which qube hosts the controller. The name of the
+    qube hosting the USB controller is displayed before the colon in the device
+    name. The string after the colon is the name of the device used within the
+    qube, like so:
 
-(Change `sys-usb` to your desired USB qube.)
+        dom0:sdb1     Cruzer () 4GiB
+        
+        usbVM:sdb1    Disk () 2GiB
 
-You can now use your USB keyboard.
+    **Note:** If your device is not listed here, you may refresh the list by
+    calling (from the qube to which the device is connected):
 
-For a confirmation dialog each time the USB keyboard is connected, change this line to:
-```
-sys-usb dom0 ask,default_target=dom0
-```
+        sudo udevadm trigger --action=change
 
-How to use a USB mouse
-----------------------
+ 3.  Assuming your USB drive is attached to dom0 and is `sdb`, we attach the
+     device to a qube with the name `personal` like so:
 
-**Caution:** Please carefully read the [Security Warning about USB Input Devices] before proceeding.
+         qvm-block a personal dom0:sdb
 
-In order to use a USB mouse, you must first attach it to a USB qube, then give that qube permission to pass mouse input to dom0.
-Edit the `qubes.InputMouse` policy file in dom0, which is located here:
+     This will attach the device to the qube as `/dev/xvdi` if that name is not
+     already taken by another attached device, or `/dev/xvdj`, etc.
 
-    /etc/qubes-rpc/policy/qubes.InputMouse
+     You may also mount one partition at a time by using the same command with
+     the partition number after `sdb`.
 
-Add a line like this to the top of the file:
+     **Warning:** when working with single partitions, it is possible to assign
+     the same partition to multiple qubes. For example, you could attach `sdb1`
+     to qube1 and then `sdb` to qube2. It is up to the user not to make this
+     mistake. The Xen block device framework currently does not provide an easy
+     way around this. Point 2 of [this comment on issue 1072][1072-comm2] gives
+     details about this.
```

Comment:
Actually, in Qubes 4.0 it is already prevented. When you attach `sdb1`, `sdb` is removed from list of available devices.